File: BehaviorPattern/Visitor/Golang/Concretes/concrete_visitor_1.go
package Concretes

import (
    "fmt"
    "visitorpattern/Interfaces"
)

// ConcreteVisitor1 is a struct that implements the Visitor interface
type ConcreteVisitor1 struct{}

// Visit implements the visit method for Elements
func (v *ConcreteVisitor1) Visit(element Interfaces.Element) {
    switch e := element.(type) {
    case *ConcreteElementA:
        fmt.Println("ConcreteVisitor1:", e.OperationA())
    case *ConcreteElementB:
        fmt.Println("ConcreteVisitor1:", e.OperationB())
    }
}

File: BehaviorPattern/Visitor/Java/Concretes/ConcreteVisitor1.java
package Concretes;

import Interfaces.Visitor;

public class ConcreteVisitor1 implements Visitor {
    @Override
    public void visit(ConcreteElementA element) {
        System.out.println("ConcreteVisitor1: " + element.operationA());
    }

    @Override
    public void visit(ConcreteElementB element) {
        System.out.println("ConcreteVisitor1: " + element.operationB());
    }
}